### PR TITLE
bpo-43439: Add audit hooks for gc functions

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue43439

### CODE REVIEW

File: Doc/whatsnew/3.10.rst [hunk 1]
```diff
@@ -676,6 +676,11 @@ encodings
 :func:`encodings.normalize_encoding` now ignores non-ASCII characters.
 (Contributed by Hai Shi in :issue:`39337`.)
 
+gc
+--
```
Comment (vstinner) [inline Doc/whatsnew/3.10.rst:681]: You may add an empty line for readability.

File: Lib/test/test_audit.py [hunk 2]
```diff
@@ -118,5 +118,19 @@ def test_socket(self):
...
+        import_helper.import_module("gc")
+        returncode, events, stderr = self.run_python("test_gc")
+        if returncode:
+            self.fail(stderr)
+
+        if support.verbose:
+            print(*events, sep='\n')
+        self.assertEqual(events[0][0], "gc.get_objects")
+        self.assertEqual(events[1][0], "gc.get_referrers")
+        self.assertEqual(events[2][0], "gc.get_referents")
```
Comment (vstinner) [inline Lib/test/test_audit.py]: You can write the the test like this, to also check the number of events:

self.assertEqual([event[0] for event in events], ["gc.get_objects", gc.get_referrers", "gc.get_referents"])

File: Lib/test/test_audit.py [hunk 3]
```diff
@@ -118,5 +118,19 @@ def test_socket(self):
         self.assertEqual(events[2][0], "socket.bind")
         self.assertTrue(events[2][2].endswith("('127.0.0.1', 8080)"))
 
+    def test_gc(self):
+        import_helper.import_module("gc")
```
Comment (vstinner) [inline Lib/test/test_audit.py]: The gc module must always be available, so I don't think that import_module() is useful to skip the test if it's missing, no?

File: Lib/test/test_audit.py [hunk 4]
```diff
@@ -118,5 +118,19 @@ def test_socket(self):
...
+            self.fail(stderr)
+
+        if support.verbose:
+            print(*events, sep='\n')
+        self.assertEqual(events[0][0], "gc.get_objects")
+        self.assertEqual(events[1][0], "gc.get_referrers")
+        self.assertEqual(events[2][0], "gc.get_referents")
+
+
+
```
Comment (vstinner) [inline Lib/test/test_audit.py:133]: nitpick: PEP 8 suggests only 2 empty lines ;-)

**Review by zooba — APPROVED** [review]
LGTM, either with or without Victor's suggestions.

This can be backported, btw. New audit events are allowed to be added in earlier releases.

**Review by vstinner — APPROVED** [review]
LGTM, thanks for the update!

Note: I don't know the format of the "audit-event::" markup in the doc, so I didn't check it.

**Pull request merged**